nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -160,9 +165,16 @@ BtNavigator::loadBehaviorTree(const std::string & bt_xml_filename)
   RCLCPP_DEBUG(get_logger(), "Behavior Tree XML: %s", xml_string.c_str());
 
   // Create the Behavior Tree from the XML input
-  tree_ = bt_->buildTreeFromText(xml_string, blackboard_);
+  tree_ = bt_->createTreeFromText(xml_string, blackboard_);
   current_bt_xml_filename_ = bt_xml_filename;
 
+  // get parameter for monitoring with Groot via ZMQ Publisher
+  if (get_parameter("enable_groot_monitoring").as_bool()) {
+    unsigned zmq_publisher_port = get_parameter("groot_zmq_publisher_port").as_int();
```

Comment:
unsigned... I think you need an `int` here, I've never actually seen just `unsigned` before.

Reply:
I'm just inheriting it from BT.CPP V3.

https://github.com/BehaviorTree/BehaviorTree.CPP/blob/cdfe773a1fcf9ec9cb84236a2c784db89b962f60/include/behaviortree_cpp_v3/loggers/bt_zmq_publisher.h#L16

Reply:
OK, don't use that. Use real types

Reply:
done